source/agora/network/Manager.d

```diff
@@ -340,32 +340,48 @@ public class NetworkManager
 
         private void getNewAddresses () nothrow
         {
-            while (1)
+            log.dbg("getNewAddresses: Start never ending loop");
+            while (this.is_running)
             {
-                scope (success)
-                    this.outer.taskman.wait(this.outer.node_config.retry_delay);
-
-                if (this.clients.empty())
-                    continue;
-
-                NetworkClient client = this.clients.front;
-                this.clients.removeFront();
-
-                try
-                {
-                    auto node_info = client.getNodeInfo();
-                    this.onNewAddresses(node_info.addresses);
-                }
-                catch (Exception ex)
+                auto peers = this.clients[].walkLength;
+                auto i = 1;
+                while (this.is_running && i <= peers)
```

Comment:
Let's make this a for loop to get rid of the temp variable?

`foreach (_; 0..this.clients[].walkLength)` ?

Reply:
Updated